### ISSUE TITLE
Delete of dangling ENIs is not executed

### ISSUE THREAD

**Issue by dosten** [weaveworks/eksctl#1856]
Hi!

Sometimes I'm unable to delete clusters because there is some dangling ENIs that prevents the stack deletion, I saw that there is some logic to remove these dangling ENIs but I don't see any info message in the output showing if it was executed, looking at the code I found this line: 

https://github.com/weaveworks/eksctl/blob/1c96f068d0d4d54bac06155417b196ad174c4079/pkg/cfn/manager/delete_tasks.go#L79

Seems like the ENIs cleanup is only executed if the stack is already failed, this is expected?

**Comment by dosten** [comment]
As a comment, after executing the deletion of the cluster the EKS cluster is properly deleted and just a subnet & VPC are keep, if I try to re run the command it fails because eksctl looks for a cluster using the EKS API instead of CloudFormation API.

Given that eksctl is unusable in this scenario I'm not sure in which case the condition I referenced above will be met.

**Comment by sayboras** [comment]
@dosten not sure if it's cascading issue of https://github.com/weaveworks/eksctl/issues/1806. What do you think ?

**Comment by dosten** [comment]
@sayboras I don't think so, the issue from #1806 I'm pretty sure is a bug introduced in #941 by mistake, what I'm saying here is that all the cleanup logic will never be executed because it will only be executed if the node group stack is already failed (at least that's what I'm seeing, I can be wrong)

**Comment by sayboras** [comment]
Understand your point now, however, there is more logic after above line to append the delete task if I am not wrong

```golang
if wait {
			tasks.Append(&taskWithStackSpec{
				info:  info,
				stack: s,
				call:  c.DeleteStackBySpecSync,
			})
		} else {
			tasks.Append(&asyncTaskWithStackSpec{
				info:  info,
				stack: s,
				call:  c.DeleteStackBySpec,
			})
		}
```

**Comment by dosten** [comment]
But that tasks will delete the nodegroups stacks, the `cleanup` function will not be executed

**Comment by sayboras** [comment]
I just got one cluster, which failed to delete (vpc and subnets are still available, i think its similar to your case), but unable to replicate issue again. Pretty much annoying as I cant recall the steps 😢

**Comment by martina-if** [comment]
Hi @dosten @sayboras do you think tihs was addressed by #1880?

**Comment by sayboras** [comment]
> do you think tihs was addressed by #1880?

@martina-if you might be right, I actually didn't have similar issue for quite sometime.

**Comment by dosten** [comment]
yeap, I also didn't have similar issues since a few months
